### ISSUE TITLE
Development solid export condition

### ISSUE THREAD

**Issue by thetarnav** [solidjs/solid-start#651]
How should library authors set up export conditions to have separate entries for dev and prod? I'm making a [tsup preset](https://github.com/solidjs-community/tsup-preset-solid) that would automatically configure those when building a library, this is just the last part that isn't working quite right.

This configuration works in the vite templates. `dev.jsx` is loaded in development and `index.jsx` in production when building.
But In solid-start the `dev.jsx` entry is taken in both cases.

Having a separate entry for the server works without issues though.
```json
{
  "solid": {
    "development": "./dist/dev.jsx",
    "import": "./dist/index.jsx"
  },
  "development": {
    "import": {
      "types": "./dist/index.d.ts",
      "default": "./dist/dev.js"
    }
  },
  "import": {
    "types": "./dist/index.d.ts",
    "default": "./dist/index.js"
  }
}
```

**Comment by ryansolid** [comment]
That is strange to say the least. I wouldn't expect a development condition showing up in prod. I believe the only place we touch this is in the vite-plugin-solid. https://github.com/solidjs/vite-plugin-solid/blob/master/src/index.ts#L327.. and in the adapters which definitely don't include `"development"`

**Comment by thetarnav** [comment]
If I switch the order of conditions (prod above dev), the prod entry is taken both in dev and prod in SolidStart.
I believe I was using the default node adapter.

**Comment by ryansolid** [comment]
I've noticed rollup node resolve plugin is kinda broken in the last 2 major versions.. can you check what version you are on. 13 seems fine. 14/15 not so much.

**Comment by thetarnav** [comment]
You are right. Overriding `@rollup/plugin-node-resolve` dependency to `13.3.0` fixed it. Solid exports are used accordingly in development/server/production.

**Comment by ryansolid** [comment]
Yeah something is really broken here just don't know what.

**Comment by aminya** [comment]
> Overriding `@rollup/plugin-node-resolve` dependency to `13.3.0` fixed it. Solid exports are used accordingly in development/server/production.

This didn't fix the issue for me

**Comment by apatrida** [comment]
Seems like ark-ui is giving up on solid-start for now due to something related to this.  
https://github.com/chakra-ui/ark/issues/1363

**Comment by ryansolid** [comment]
In setting up for SolidStarts next Beta Phase built on Nitro and Vinxi we are closing all PRs/Issues that will not be merged due to the system changing. If you feel your issue was closed by mistake. Feel free to re-open it after updating/testing against 0.4.x release. Thank you for your patience.  See https://github.com/solidjs/solid-start/pull/1139 for more details.

**Comment by ryansolid** [comment]
I'm going to transfer this one to docs instead. General how to make a package for SSR is good thing to have.
